### DISCUSSION TITLE
Buffer vs. Document

### DISCUSSION THREAD

**Discussion by A-Walrus** [helix-editor/helix#4039] [General] [answered]
I believe these two terms are interchangeable, but I'm not sure (that's part of the problem). In the documentation, the term buffer is mostly used, but document is sometimes used as well:

<details> <summary>Usage of the term document in the book</summary>

```
usage.md
148:document and a special tree-sitter query file to work properly. [Only

themes.md
225:| `ui.statusline.inactive`  | Statusline (unfocused document)                |

configuration.md
163:documents with this language.

keymap.md
264:| `s`     | Open document symbol picker (**LSP**)                                   | `symbol_picker`                     |
266:| `g`     | Open document diagnostics picker (**LSP**)                              | `diagnostics_picker`                |
299:| `[D`     | Go to first diagnostic in document (**LSP**) | `goto_first_diag`     |
300:| `]D`     | Go to last diagnostic in document (**LSP**)  | `goto_last_diag`      |

generated/typable-cmd.md
19:| `:line-ending` | Set the document's default line ending. Options: crlf, lf. |

```
</details>

Within the code the term document is more often used, and buffer usually refers to a TUI text buffer.

Also, looking at the code for `buffer_close` for example, we see that it is dealing with `document_ids` not `buffer_ids`, which is a bit confusing.
```rust
fn buffer_close(
    cx: &mut compositor::Context,
    args: &[Cow<str>],
    event: PromptEvent,
) -> anyhow::Result<()> {
    if event != PromptEvent::Validate {
        return Ok(());
    }

    let document_ids = buffer_gather_paths_impl(cx.editor, args);
    buffer_close_by_ids_impl(cx.editor, &document_ids, false)
}
```

So... do they mean the same thing, in which case we should probably do some renaming for consistency, or are they separate terms, in which case there should be some documentation explaining the difference.

**Comment by archseer** [comment, marked as answer]
They're the same thing and in code I thought document made more sense. But for commands and user documentation everyone expects "buffer"
